### PR TITLE
HE pipe turf interaction

### PR DESCRIPTION
**Pull request by TheLordME**
Yes hello, I was inspired by the water turf under the HE pipes to make this PR.
I have implemented exactly these changes on CitRP, and its probably the best way to do it zone bases atmos.

I know that its my master branch, whoopsie on my end, havent tested it localy, but as I said, it works fine on CitRP, where it has been full merged for some time now, without any major issues.

the relavant PRs are:
https://github.com/Citadel-Station-13/Citadel-Station-13-RP/pull/3614
and
https://github.com/Citadel-Station-13/Citadel-Station-13-RP/pull/3651

### CODE REVIEW

**Review by Atermonera — DISMISSED** [review]

File: code/ATMOSPHERICS/pipes/he_pipes.dm [hunk 1]
```diff
@@ -98,7 +98,7 @@
 			else
 				var/datum/gas_mixture/environment = loc.return_air()
 				environment_temperature = environment.temperature
-			if(abs(environment_temperature-pipe_air.temperature) > minimum_temperature_difference)
+			if((abs(environment_temperature-pipe_air.temperature) > minimum_temperature_difference) || (location_as_turf.special_temperature))
```
Comment (Spookerton) [inline code/ATMOSPHERICS/pipes/he_pipes.dm:101]:
```suggestion
			var/turf/loc_as_turf = loc
			if((abs(environment_temperature-pipe_air.temperature) > minimum_temperature_difference) || (loc_as_turf.special_temperature))
```

**Review by Spookerton — CHANGES_REQUESTED** [review]

File: code/ATMOSPHERICS/pipes/he_pipes.dm [hunk 2]
```diff
@@ -98,7 +98,8 @@
 			else
 				var/datum/gas_mixture/environment = loc.return_air()
 				environment_temperature = environment.temperature
-			if(abs(environment_temperature-pipe_air.temperature) > minimum_temperature_difference)
+			var/turf/loc_as_turf = loc
```
Comment (Spookerton) [inline code/ATMOSPHERICS/pipes/he_pipes.dm:101]:
```suggestion
			var/turf/simulated/loc_as_turf = loc
```
My bad. I didn't check what level you defined special_temperature.

**Review by Spookerton — CHANGES_REQUESTED** [review]

**Comment by Spookerton** [comment]
@TheLordME Beep.

**Comment by TheLordME** [comment]
Gonna push the changes in ~ 3 hours

**Review by Spookerton — APPROVED** [review]

**Comment by Spookerton** [comment]
Code do the thing. @Atermonera had an approval on the prior failing code; I assume that's conceptual.

**Pull request merged**